### PR TITLE
Add remote-redux-devtools store enhancer

### PR DESCRIPTION
**Pull request by clauderic**
See https://github.com/zalmoxisus/remote-redux-devtools

### CODE REVIEW

**Comment by emmby** [comment]
This should be limited to dev builds only

**Comment by clauderic** [comment]
Ah, yes. Good point! Updated PR :)

File: src/redux/store.js [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
 import reducer from './reducer';
 
-const enhancer = compose(
+let enhancers = [
   applyMiddleware(...middleware),
   reduxLoop.install()
-);
+];
+
+if (process.env.NODE_ENV !== 'production') {
```
Comment (jevakallio) [inline src/redux/store.js]: How about using `if (__DEV__)` here? That's the convention we typically use, and keeping a single way of detecting dev builds makes it easier to write tooling around this in the future.

**Comment by jevakallio** [comment]
@clauderic this looks great, tested the dev tools and they worked great.

I have a couple of questions before merging this though.

1. Does the dev tool send all Redux dispatches to the remotedev.io service even when I am not connected to remotedev.io myself via a browser?

2. Are there any other issues around developers who don't use the remote redux dev tools? This tool looks extremely useful and I'll likely start using it myself from now on, but I don't want to force this upon all our users. 

3. I couldn't get the Atom plugin to connect though, any clue why?

**Comment by clauderic** [comment]
Regarding 1 and 2, I'm not sure, though I can look into it. Perhaps @zalmoxisus would care to chime in?

As for 3, I've experienced the same thing, the Atom tools don't currently seem to support the remote feature.

**Comment by zalmoxisus** [comment]
1. It will send the history to the remote monitor only when you open the monitor (`remotedev.io/local` in the browser, Redux DevTools extension or any other apps indicated there). When you close the monitor it will not send anything, so will not affect the performance significantly (as the serialization is the bottle neck there). And yes, you can use [a local server](https://github.com/zalmoxisus/remotedev-server), which you can [start together with your dev server](https://github.com/zalmoxisus/remotedev-server#import-in-your-serverjs-script-you-use-for-starting-a-development-server). Or if you use [React Native Debugger](https://github.com/jhen0409/react-native-debugger), it will start the local server for you.

2. As long as nobody opens the monitor, there would be no issues. When someone open the monitor, he will get the logs from all instances of all developers and even can synchronize them and dispatch some actions remotely. If you don't want that, set [`realtime` parameter](https://github.com/zalmoxisus/remote-redux-devtools#parameters) to `false` and use a specific action to start the monitoring or just to send logs as in [this example](https://github.com/zalmoxisus/remote-redux-devtools/blob/master/examples/toggle-monitoring/store/configureStore.js#L12-L14) ([demo](http://zalmoxisus.github.io/monitoring/)).

3. Most likely the `remotedev.io` server was overloaded. There's no `Settings` button available as reported [here](https://github.com/zalmoxisus/atom-redux-devtools/issues/3), so no way to use local server for now. I will fix it as soon as get [`remotedev-app`](https://github.com/zalmoxisus/remotedev-app) `0.5` released.

**Comment by jevakallio** [comment]
@clauderic the comments above sound good to me, we could merge this in. The branch has some conflict with the latest changes, would you have a moment to rebase from master and update your branch?

**Comment by jhabdas** [comment]
I tried remote dev tools about a month and they worked well, when they worked at all. I was at a coffee shop when I was developing and got annoyed enough with the remote aspect I tried setting up locally but that was more complex than I had a stomach or time for. One thing of note is that if any sensitive information (e.g. PII) flows through these dev tools there's no telling where it'll end up or who might be able to see it.

**Comment by clauderic** [comment]
I've got it working with localhost, it's fairly simple. I'll update my PR as soon as I get the chance with the localhost configs. 

The only thing I still need to do is set up the remote-devtools server to start automatically on `npm start`, assuming we want it enabled by default (could also be conditional, runtime flag perhaps). 
Thoughts @jevakallio?

**Comment by zalmoxisus** [comment]
@clauderic, [`react-native-debugger`](https://github.com/jhen0409/react-native-debugger) app already includes and starts `remotedev-server` automatically. Another option is to use [`remote-redux-devtools-on-debugger`](https://github.com/jhen0409/remote-redux-devtools-on-debugger), which also already includes `remotedev-server`.

**Comment by tino-junge** [comment]
@clauderic finally revisited now your PR :) After I updated redux-dev-tools I had to make some changes to get it working again (now using React Native Debugger). Couldn't apply my changes directly to your repo so I've opened a new PR #183 for it. We can close this PR here once we merge the other one.

**Comment by krivachy** [comment]
Merged #183 so closing this.

**Pull request merged**